src/MongoDB.Driver.Core/Core/Operations/BulkWriteConcernError.cs

```diff
@@ -56,11 +55,25 @@ public BulkWriteConcernError(int code, string message, BsonDocument details)
         /// <param name="message">The message.</param>
         /// <param name="details">The details.</param>
         public BulkWriteConcernError(int code, string codeName, string message, BsonDocument details)
+            : this(code, codeName, message, details, new string[0])
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BulkWriteConcernError" /> class.
+        /// </summary>
+        /// <param name="code">The code.</param>
+        /// <param name="codeName">The name of the code.</param>
+        /// <param name="message">The message.</param>
+        /// <param name="details">The details.</param>
+        /// <param name="errorLabels">The error labels.</param>
+        public BulkWriteConcernError(int code, string codeName, string message, BsonDocument details, IEnumerable<string> errorLabels)
         {
             _code = code;
             _codeName = codeName;
             _details = details;
             _message = message;
+            _errorLabels = Ensure.IsNotNull(errorLabels, nameof(errorLabels));
```

Comment:
Note: `errorLabels` is never `null` in our code, so the `Ensure` here is for double checking that consumer doesn't break other code which uses this array up the call stack.